### PR TITLE
replace Travis with Github (actions) CI

### PR DESCRIPTION
**Pull request by temoto**
I think the only thing that's not working are database tests. Mysql and postgresql start up correctly, but python can't connect to them. Will continue later.

### CODE REVIEW

**Comment by temoto** [comment]
Why this change is useful: builds on Travis take about 75 minutes, on Github - less than 5 minutes to useful result, 10 minutes to timeout of pypy jobs.

The only failing check is code coverage -1%.

File: .github/workflows/style.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  style:
+    runs-on: ubuntu-latest
+    # https://github.community/t/duplicate-checks-on-push-and-pull-request-simultaneous-event/18012/5
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != 'eventlet/eventlet'
+    timeout-minutes: 5
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: setup python ${{ matrix.python-version }}
```
[the remark this answers is not in the document]
Comment (temoto) [reply .github/workflows/style.yaml:14]: Thanks, I copy-pasted it mindless.

File: .github/workflows/style.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
```
[the remark this answers is not in the document]
Comment (temoto) [reply .github/workflows/style.yaml:17]: Yeah it's just any Python to run tox.

File: .travis.yml [hunk 3]
```diff
@@ -1,68 +0,0 @@
...
-    - {python: 3.8, env: TOXENV=py38-epolls}
-    - {python: 3.8, env: TOXENV=py38-poll}
-    - {python: 3.8, env: TOXENV=py38-selects}
-
-    - {python: pypy, env: TOXENV=pypy-epolls}
-    - {python: pypy, env: TOXENV=pypy-poll}
-    - {python: pypy, env: TOXENV=pypy-selects}
-
-    - {python: 3.6, env: TOXENV=ipv6}
-    - {python: 3.8, env: TOXENV=py38-openssl}
```
[the remark this answers is not in the document]
Comment (temoto) [reply .travis.yml:35]: No, it is my mistake. Thanks.

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  schedule:
+    - cron: "43 7 */14 * *" # every two weeks, time chosen by RNG
+jobs:
+  tox:
+    name: "tox ${{ matrix.toxenv }}"
+    continue-on-error: ${{ matrix.ignore-error }}
+    runs-on: ubuntu-latest
+    # https://github.community/t/duplicate-checks-on-push-and-pull-request-simultaneous-event/18012/5
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != 'eventlet/eventlet'
+    timeout-minutes: 10
```
[the remark this answers is not in the document]
Comment (temoto) [reply .github/workflows/test.yaml:14]: In that particular test run you linked, `test_visibility_from_other_connections` passed.

And that was sporadic failure, passed when I re-run jobs.

**Review by tipabu — APPROVED** [review]

**Comment by jstasiak** [comment]
> While I'm not a fan of bundling unrelated (at least seemingly) changes (like upgrading dependencies) with the changes related to the task at hand – I don't have time to do it better. :)

And, in case I'm misunderstood, I didn't mean to imply anything derogatory about your work. It's a very nice change and a lot of effort put into it!

**Pull request merged**